app/containers/AccountProvider/saga.js

```diff
@@ -165,6 +180,58 @@ export function* isAvailableAction(isValid) {
   yield call(isValid);
 }
 
+const getReferralInfo = async (user, eosService) => {
+  const info = await eosService.getTableRow(
+    INVITED_USERS_TABLE,
+    INVITED_USERS_SCOPE,
+    user,
+    process.env.EOS_TOKEN_CONTRACT_ACCOUNT,
+  );
+  return info;
+};
+
+function* updateRefer(user, eosService) {
+  const receivedCookieName = `${REFERRAL_REWARD_RECEIVED}_${user}`;
+  const noInviterCookieName = `${NO_REFERRAL_INVITER}_${user}`;
+  if (getCookie(receivedCookieName) || getCookie(noInviterCookieName)) {
+    return;
+  }
+
+  const info = yield call(getReferralInfo, user, eosService);
+
+  if (info) {
+    const reward = +convertPeerValueToNumberValue(info.common_reward);
+    if (reward) {
+      setCookie({ name: receivedCookieName, value: true });
+      yield put(
+        addToast({
+          type: 'success',
+          text: 'You received referral reward!',
+        }),
+      );
+    }
+  } else {
+    setCookie({
+      name: noInviterCookieName,
+      value: true,
+    });
+  }
+}
+
+const rewardRefer = async (user, eosService) => {
+  try {
+    await eosService.sendTransaction(
+      user,
+      REWARD_REFER,
+      {
+        invited_user: user,
+      },
+      process.env.EOS_TOKEN_CONTRACT_ACCOUNT,
+      null,
+    );
+  } catch (err) {}
```

Comment:
The error should be should be reported to our backend.

If the operation failed, then cookies should not be set and the operation attempted again during the next profile update.

Reply:
Ok, added